Feature/mark referrers as spam

marco-altran opened the pull request:
Fixes #11818 

Depends on: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1704

This PR adds the possibility to mark and unmark a referrer as spam by accessing the `reportReferrerAsSpam` and `unreportReferrerAsSpam` methods from `ReferrersStore`. The functionally was added through a menu popup that is activated on long press:

![wordpress-referrer-3](https://user-images.githubusercontent.com/69607277/94866037-fcad3980-0414-11eb-8933-fe24443080d2.gif)

The desired state is that when marked as spam, the referrer is greyed out and the user still has the ability to unmark as spam. If he does a pull to refresh, the referrer disappears (except if it's a search engine where the WP API ignores the report). This is very similar to the existing behavior in wordpress.com.

To test:

Go to Stats -> select the period (select YEARS for maximum referrals) -> go to referrers -> long press -> mark/unmark as spam

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

planarvoid committed:
Add stock media source

marecar3 committed:
Update gb-mobile reference

antonis committed:
Retrieves selected layout content on create page action

antonis committed:
Passes selected layout content to the editor

antonis committed:
Loads selected layout content to the editor

planarvoid committed:
Remove get method and add stock media source

antonis committed:
Merge branch 'issue/2443-MLP_WPCOM' into issue/2446-MLP_CreatePage

malinajirka committed:
Update targetSdkVersion to v29 and fix breaking changes

malinajirka committed:
Update WPStories reference

malinajirka committed:
Merge branch 'develop' into update-target-sdk-to-29

mchowning committed:
Add message when cannot load mentions because offline

mchowning committed:
Update release notes

malinajirka committed:
Merge branch 'develop' into update-target-sdk-to-29

malinajirka committed:
Use SDK 29 on CI

malinajirka committed:
Update compile and target sdk in dependencies

malinajirka committed:
Suppress false positive lint issue

malinajirka committed:
Update gutenberg-mobile ref

oguzkocer committed:
Adds FluxC as composite build

planarvoid committed:
Merge branch 'develop' into feature/12641-add-pexel-source-to-media-picker

antonis committed:
Merge branch 'issue/2443-MLP_WPCOM' into issue/2446-MLP_CreatePage

antonis committed:
Merge branch 'issue/2443-MLP_WPCOM' into issue/2446-MLP_CreatePage

antonis committed:
Sets correct font family

planarvoid committed:
Update stock media data source with the latest fluxC version

marecar3 committed:
no message

develric committed:
Refactor to make the variable onlyImagesSelected more meaningful.

aforcier committed:
Update stories submodule reference

aforcier committed:
Implement StoriesAnalyticsListener

planarvoid committed:
Merge branch 'develop' into feature/12641-add-pexel-source-to-media-picker

malinajirka committed:
Update gutenberg-mobile ref

planarvoid committed:
Add progress dialog when stock media are being uploaded

planarvoid committed:
Update fluxC version

planarvoid committed:
Remove browser type from the MediaPickerActivity.kt

planarvoid committed:
Fix tests

planarvoid committed:
Move feature image tracking to EditPostSettingsFragment

planarvoid committed:
Add new method to LiveDataUtils.kt

antonis committed:
Merge branch 'issue/2443-MLP_WPCOM' into issue/2446-MLP_CreatePage

planarvoid committed:
Only trigger featured image pick tracking once

planarvoid committed:
Merge branch 'feature/remove-browser-type-from-media-picker' into feature/12641-add-pexel-source-to-media-picker

develric committed:
Merge branch 'develop' into fix/media-picker-variable-refactor

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/mediapicker/MediaPickerViewModel.kt

develric committed:
Reviewing badge margins.

develric committed:
Adding tooltip to edit icon.

antonis committed:
Removes unneeded code

antonis committed:
Aligns back arrow to the left

planarvoid committed:
Handle featured image upload

antonis committed:
Removes bottom shadow extra space

antonis committed:
Adds white background in selection checkmark

antonis committed:
Fixes top toolbar dimension issues

antonis committed:
Merge branch 'develop' into issue/2446-MLP_CreatePage

antonis committed:
Sets rounded top corners in non expanded modal state

antonis committed:
Removes bottomsheet default dim

marecar3 committed:
Present jetpack security settings and send switch condition

marecar3 committed:
Merge branch 'develop' into gutenberg/unsupported-block-editor-switch-to-enable-jetpack-sso

malinajirka committed:
Update Readme after the migration to JDK11

planarvoid committed:
Move local media tracking to the DeviceListBuilder.kt

planarvoid committed:
Remove unused NoChange state from media source

planarvoid committed:
Remove unused stock media upload from MediaPickerActivity.kt

planarvoid committed:
Add default search view flag to MediaPickerSetup.kt

planarvoid committed:
Add default search view and multiple selection flags to MediaPickerLauncher.kt

planarvoid committed:
Fix logic to show and select stock media

planarvoid committed:
Update FluxC version

planarvoid committed:
Fix lint issues

antonis committed:
Revert "Sets rounded top corners in non expanded modal state"

This reverts commit 7c0ab38c06d36f210dcf671ad1f4b12af04f96f7.

antonis committed:
Sentence casing on the heading

planarvoid committed:
Merge pull request #13004 from wordpress-mobile/fix/media-picker-variable-refactor

Refactor to make variable more in line with it's meaning.

mchowning committed:
Merge branch 'develop' into add_offline_message_for_mentions

mchowning committed:
Merge pull request #12968 from wordpress-mobile/add_offline_message_for_mentions

Add message when cannot load mentions because offline

malinajirka committed:
Update gutenberg ref

malinajirka committed:
Merge branch 'develop' into update-target-sdk-to-29

malinajirka committed:
Merge pull request #12947 from wordpress-mobile/update-target-sdk-to-29

Update target and compile sdk version to v29

planarvoid committed:
Create new request code for stories picker to fix previous wrong behaviour

antonis committed:
Header area spacing improvements

planarvoid committed:
Fix story creation

planarvoid committed:
Fix lint error

planarvoid committed:
Fix unit tests

planarvoid committed:
Add tests for StockMediaDataSource.kt

planarvoid committed:
Only track the `IMAGE_PICKED` event when the media picker is enabled

marecar3 committed:
Update gb-mobile reference

antonis committed:
Header spacing improvements

antonis committed:
Merge branch 'develop' into issue/2446-MLP_CreatePage

marecar3 committed:
Update capabilities when user leaves jetpack settings screen

marecar3 committed:
Update gb-mobile reference

marecar3 committed:
Merge branch 'develop' into gutenberg/unsupported-block-editor-switch-to-enable-jetpack-sso

marecar3 committed:
Update gb-mobile reference

jkmassel committed:
Merge pull request #13013 from wordpress-mobile/update-readme

Update Readme after the migration to JDK11

etoledom committed:
Update gutenberg ref

marecar3 committed:
Remove full package id

marecar3 committed:
Move boolean cast before if else block

marecar3 committed:
Merge branch 'develop' into gutenberg/unsupported-block-editor-switch-to-enable-jetpack-sso

marecar3 committed:
Update gb-mobile reference

oguzkocer committed:
Improves how FluxC is included as a composite build

oguzkocer committed:
Replaces fluxCDependency name with fluxCBinaryPath

marecar3 committed:
Fix regression which was brought by renaming var from unsupportedBlockEditorSwitch to canEnableUnsupportedBlockEditor

antonis committed:
Moved create page handling to Viewmodel

etoledom committed:
Update gutenberg ref

etoledom committed:
Update gutenberg ref

etoledom committed:
Merge pull request #13011 from wordpress-mobile/gutenberg/unsupported-block-editor-switch-to-enable-jetpack-sso

[Gutenberg] UBE: Pressenting Jetpack Security settings on Missing Block

loremattei committed:
Add missing strings for screenshots

loremattei committed:
Merge pull request #13020 from wordpress-mobile/fix/add-missing-screenshots-strings

Add missing strings for screenshots

cameronvoell committed:
Release script: Update gutenberg-mobile ref

cameronvoell committed:
Release script: Update strings

renanferrari committed:
Bump default LoginFlow's FluxC version

khaykov committed:
Removed unused values.

khaykov committed:
Updated My Site layout.

khaykov committed:
Updated release notes.

khaykov committed:
Merge pull request #13008 from wordpress-mobile/fix/media-picker-edit-button-tweaks

Media picker edit button tweaks

malinajirka committed:
Merge pull request #12943 from wordpress-mobile/issue/2446-MLP_CreatePage

[Modal Layout Picker] "Create Page" button functionality

khaykov committed:
Fixed about activity color.

malinajirka committed:
Squashed 'libs/analytics/' changes from 3a16fd3bb3..aac18bf1c8

aac18bf1c8 Bump version to 1.3.1
1e18ac220d Merge commit '0411566cd9245d5044d16b7401fbf73024aea60f' into update-analytics-lib
7f13f20a3c Merge branch 'develop' into issue/6700-login-progress-dialog-forever
b0a8f68664 Add Tracks of the service phase updates
0f6184eca8 Update analytics and network library build files
f388b5308c Merge pull request #6713 from wordpress-mobile/issue/6389-upload-service-notification
bb2b9a12e5 Add analytics events for login prologue view pager
bf59e4a2ce fixed merge conflicts
4bf8047b5c Login: Overlooked a few references that needed removing.
77970f7148 Login: Removes redundant analytics.
401d529a59 Merge pull request #6811 from wordpress-mobile/mzorz/analytics-write-post-retry
d531efb9d1 Login: Fix analytics key.
00eb2c8045 Login: remove analytic keys we won't use for now.
88bc0c45e2 Login: Adds social login analytics keys.
ed54e422c9 added analytics tracking for WRITE POST and RETRY quick actions
8aac2c262e Add analytics to social login logged in method
bd69be0018 Add analytics to track Google login
8e831f35e0 Bump analytics when the user share something to WP with success

git-subtree-dir: libs/analytics
git-subtree-split: aac18bf1c8b529899794ee7613d4ce64665b41dc

malinajirka committed:
Merge commit '78f990952aa0ffaaff76453487b2cbd0e1256e07' into update-wp-analytics

loremattei committed:
Merge pull request #13021 from wordpress-mobile/release/15.8

Merge Release/15.8 into develop

MarcoAltranDC committed:
WIP: Mark referrer as spam

cameronvoell committed:
Updating aztec reference to tagged release v1.3.45

MarcoAltranDC committed:
WIP: Improved dialog

marco-altran committed:
WIP: Fix item click

zwarm committed:
Add check for instagram image click

renanferrari committed:
Merge pull request #13028 from wordpress-mobile/issue/fix-about-app-bg

Fixed about activity color

zwarm committed:
Minor cleanup

renanferrari committed:
Remove unused method

renanferrari committed:
Introduce email parameter for enterEmailAddress method

renanferrari committed:
Introduce password parameter for enterPassword method

zwarm committed:
Check for ig_embed

renanferrari committed:
Add passwordless user email property to gradle.properties-example

renanferrari committed:
Add new login test for passwordless users

renanferrari committed:
Add mocked response for login with passwordless user

khaykov committed:
Merge pull request #13024 from wordpress-mobile/loginflow-fluxc-version-bump

Bump default LoginFlow's FluxC version

khaykov committed:
Merge pull request #13033 from wordpress-mobile/add-passwordless-flow-test

Login: Add test case for the passwordless login flow

khaykov committed:
Adjusted avatar alignment.

khaykov committed:
Realigned views, fixed issue with long quick action button labels messing with layout.

khaykov committed:
Fixed lint error.

mzorz committed:
removed the contentResolver.getMimeType - isSupportedMimeType() check given it was taking too long for devices with many media files

mokagio committed:
Make FluxC a composite build dependency (#13001)

planarvoid committed:
Merge branch 'feature/remove-browser-type-from-media-picker' into feature/12641-add-pexel-source-to-media-picker

planarvoid committed:
Extract insert logic into Insert use cases

develric committed:
Merge pull request #13035 from wordpress-mobile/issue/12996-delay-media-picker-quick-fix

Removed the `contentResolver.getMimeType - isSupportedMimeType() `  check for populating the picker

planarvoid committed:
Fix lint issue

loremattei committed:
Bump version number

zwarm committed:
Wrap verse block

develric committed:
Merge pull request #13007 from wordpress-mobile/feature/remove-browser-type-from-media-picker

Remove browser type from media picker

marco-altran committed:
WIP: Long item click show menu popup with Open Website and Mark as Spam

Click opens website

marco-altran committed:
WIP: cleaning up unused methods, classes and resources

loremattei committed:
Fix access to the prerelease param

cameronvoell committed:
Updated gutenberg ref

marco-altran committed:
Added actions for Mark as Spam and Open websites. Added tracking events

cameronvoell committed:
Updated gutenberg ref

marco-altran committed:
Data refresh upon marking domain as spam

zwarm committed:
Add gridline color for table block

khaykov committed:
Increased the size of quick action container.

marco-altran committed:
Upgraded FluxC and improved spam mark

marco-altran committed:
Fixed mark spam on groups

Improved refactor and add light grey on spam

marco-altran committed:
Completed unmark as spam feature

marco-altran committed:
Improved unit tests and some refactoring

marco-altran committed:
Merge pull request #1 from wordpress-mobile/develop

Rebase

marco-altran committed:
Merge branch 'develop' into feature/mark-referrers-as-spam

# Conflicts:
#	WordPress/build.gradle

marco-altran committed:
Rolled back FluxC to the fork's version and added change in RELEASE-NOTES.txt

jkmassel committed:
Merge pull request #13039 from wordpress-mobile/release/15.8

Merge Release/15.8 back into develop

jkmassel committed:
Merge pull request #13040 from wordpress-mobile/fix/set-prerelease-flag-on-github

Set the prerelease flag on GitHub release drafts

cameronvoell committed:
Merge branch 'develop' into gutenberg/integrate_release_1.38.0

ceyhun committed:
Revert "Release script: Update strings"

This reverts commit eacf95cbf97132ec11dc52c6830e1576b083e024.

ceyhun committed:
Move 'Autogenerated:Gutenberg Native' mark to correct place

ceyhun committed:
Update strings

loremattei committed:
Updates translations

loremattei committed:
Bump version number

loremattei committed:
Update metadata translations for 15.8

ceyhun committed:
Update RELEASE-NOTES.txt

ashiagr committed:
Merge pull request #13038 from wordpress-mobile/issue/13037-wrap-verse-block

Wrap verse block

ashiagr committed:
Merge pull request #13041 from wordpress-mobile/issue/11771-reader-table-render

Add gridline color for table block

cameronvoell committed:
Update RELEASE-NOTES.txt

Updated release notes with shadow dimensions crash fix

loremattei committed:
Fix translations

cameronvoell committed:
Update RELEASE-NOTES.txt

Added release notes for accessibility mode improvement

cameronvoell committed:
Updated gutenberg ref to release on trunk

cameronvoell committed:
Merge pull request #13023 from wordpress-mobile/gutenberg/integrate_release_1.38.0

Integrate gutenberg-mobile release 1.38.0

khaykov committed:
Merge branch 'develop' into issue/fix-site-picker-layout

develric committed:
Merge pull request #13026 from wordpress-mobile/issue/fix-site-picker-layout

Issue/fix my site layout on bigger screens

renanferrari committed:
Update behavior when handling Jetpack sites so it uses site credentials

marco-altran committed:
Icon bugfix when unmarking as spam

malinajirka committed:
Merge pull request #13032 from wordpress-mobile/issue/12767-instagram-embed

Issue/12767 instagram embed

planarvoid committed:
Merge branch 'develop' into feature/12641-add-pexel-source-to-media-picker

loremattei committed:
Merge pull request #13050 from wordpress-mobile/release/15.8

Merge Release/15.8 back into develop

loremattei committed:
Bump version number

loremattei committed:
Update draft release notes for 15.9.

loremattei committed:
Update release notes.

loremattei committed:
Send strings to translation.

planarvoid committed:
Fix crash when switching to empty list from a list of results

loremattei committed:
Update some items to use relative paths

planarvoid committed:
Fix dialog leaking the activity context

malinajirka committed:
Merge pull request #13053 from wordpress-mobile/fix/lint-error-on-local-builds

Update some items to use relative paths

planarvoid committed:
Fix lint error

aforcier committed:
Merge pull request #13029 from wordpress-mobile/update-wp-analytics

Sync with Analytics subtree

marco-altran committed:
Updated FluxC library

planarvoid commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/StatsListViewModel.kt:
```diff
@@ -27,6 +27,7 @@ import org.wordpress.android.ui.stats.refresh.lists.StatsListViewModel.StatsSect
 import org.wordpress.android.ui.stats.refresh.utils.ItemPopupMenuHandler
 import org.wordpress.android.ui.stats.refresh.utils.NewsCardHandler
 import org.wordpress.android.ui.stats.refresh.utils.StatsDateSelector
+import org.wordpress.android.ui.stats.refresh.utils.StatsDateSelector.Factory
```
I think I'd revert this change. It removes a little bit of readability and you're not touching this file for any other reason.

marco-altran replied:
I did some modifications in this class that was later rolled back. I kept this one because it solved an Android Studio warning. I am fine with both, please let me know what you think it's best.

marco-altran replied:
Fixed in [af23b7e](https://github.com/wordpress-mobile/WordPress-Android/pull/13047/commits/af23b7e98f96188ee4c047c25e92eb9f8d586c3c)

planarvoid commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/BlockListItem.kt:
```diff
@@ -124,6 +124,7 @@ sealed class BlockListItem(val type: Type) {
         val showDivider: Boolean = true,
         val textStyle: TextStyle = TextStyle.NORMAL,
         val navigationAction: NavigationAction? = null,
+        val menuAction: ((View) -> Boolean)? = null,
```
I know it's not consistent with the `navigationAction` but I think I'd call this one `longClickAction` and (and that's for another PR) I'd rename the `navigationAction` to something like `clickAction`. The reason is that the UI doesn't need to know that the action navigates somewhere or opens a menu. It's just necessary to know what user action should trigger it. Does it make sense? However, this is not super important :-)

marco-altran replied:
I totally agree, it was super confusing to me that the `navigationAction` actually is the `onItemClick` I was looking for. I will make this change.

marco-altran replied:
Fixed in [af23b7e](https://github.com/wordpress-mobile/WordPress-Android/pull/13047/commits/af23b7e98f96188ee4c047c25e92eb9f8d586c3c)

planarvoid commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/granular/usecases/ReferrersUseCase.kt:
```diff
@@ -180,7 +193,12 @@ constructor(
+    private fun buildTextStyle(spam: Boolean) = if (spam) LIGHT else TextStyle.NORMAL
+
+    private fun buildIcon(iconUrl: String?, spam: Boolean): Int? {
+        if (spam) {
```
I think I'd merge this condition into the `when` bellow into something like:
```
when {
spam -> ...
iconUrl == null -> ...
...
}

marco-altran replied:
Good thinking. I will make the change

marco-altran replied:
Fixed in [af23b7e](https://github.com/wordpress-mobile/WordPress-Android/pull/13047/commits/af23b7e98f96188ee4c047c25e92eb9f8d586c3c)

planarvoid commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/viewholders/BlockListItemViewHolder.kt:
```diff
@@ -49,6 +49,7 @@ open class BlockListItemViewHolder(
                 imageManager.load(this, item.icon)
             }
             item.iconUrl != null -> {
+                this.imageTintList = null
```
👍

planarvoid commented on libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java:
```diff
@@ -102,6 +102,12 @@
         STATS_REFERRERS_ITEM_TAPPED,
+
+        // FIXME Verify p2 post before adding these stats
+        STATS_REFERRERS_ITEM_LONG_PRESSED,
```
Make sure you also add the mapping to the correct value to `org.wordpress.android.analytics.AnalyticsTrackerNosara#getEventNameForStat`. Usually we just map the event to lowercase string.

marco-altran replied:
Got it, I will make this change.

marco-altran replied:
Fixed in [af23b7e](https://github.com/wordpress-mobile/WordPress-Android/pull/13047/commits/af23b7e98f96188ee4c047c25e92eb9f8d586c3c)

planarvoid reviewed (commented):
It's looking good and works really well! I have a few comments for the code.

Looking at the WordPress.com functionality, I think it would be better if the `mark as spam` option was enabled on the group of referrers instead the children. If you mark a child of a group as a spam, the whole group disappears anyway. What do you think? We can have it enabled also on children but I think it makes sense to have it on the parent. 

I was trying to find out how the `Search Engines` and the `Android App` items are disabled for the `mark as spam` feature in WordPress.com and from the response I have no idea. I'm expecting some kind of a hack.

planarvoid committed:
Only show progress dialog when an item takes longer to load

loremattei committed:
Merge pull request #13054 from wordpress-mobile/release/15.9

Merge Release/15.9 back into develop

planarvoid committed:
Add support for snackbar errors to media picker

planarvoid committed:
Fix unit test

planarvoid committed:
Ask for both write and read storage permissions in the media picker

planarvoid committed:
Fix test

renanferrari committed:
Update release notes to account for new Jetpack site login behavior

marco-altran committed:
Added analytics and minor refactoring

marco-altran committed:
Merge pull request #2 from wordpress-mobile/develop

Rebase

marco-altran committed:
Merge branch 'develop' into feature/mark-referrers-as-spam

marco-altran committed:
Fixed RELEASE-NOTES.txt merge

marco-altran committed:
Added mark as spam to groups instead of the children

marco-altran commented:
> It's looking good and works really well! I have a few comments for the code.
> 
> Looking at the WordPress.com functionality, I think it would be better if the `mark as spam` option was enabled on the group of referrers instead the children. If you mark a child of a group as a spam, the whole group disappears anyway. What do you think? We can have it enabled also on children but I think it makes sense to have it on the parent.

Fixed with [5c226f6](https://github.com/wordpress-mobile/WordPress-Android/pull/13047/commits/5c226f6825ec1e0e7ec158f2e72e0678cbcee86f) and [9b4893e in FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1704/commits/9b4893e13db6571a9116d5748d910d5e51ea9874)

Keep in mind that the functionality for marking a child as spam still exists, I just deactivated through the removal of `onMenuAction`. It will be easier to change the behavior if we want to, but I am happy to completely rollback these changes if you think it's better that way.

> I was trying to find out how the `Search Engines` and the `Android App` items are disabled for the `mark as spam` feature in WordPress.com and from the response I have no idea. I'm expecting some kind of a hack.

I was wrong in thinking that the behavior was caused by some groups having an URL and others don't. Indeed we have groups with and without URLs, but it has nothing to do but it had no correlation with the behavior. I noticed empirically that all groups that can be marked as spam can be resolved to a valid URL, so I added a validation using `UrlUtils.isValidUrlAndHostNotNull("https://${group.name}")` and a small modification in FluxC to replicate this behavior. It's a hack, but it's working exactly like in wordpress.com. Let me know what you think about it.

ashiagr committed:
Default to (Untitled) on the reader post card when blogName is empty

ashiagr committed:
Remove unused import

ashiagr committed:
Make blogName non-nullable

malinajirka committed:
Merge pull request #13071 from wordpress-mobile/issue/fix-layout-issue-blank-site-title

Default to (Untitled) on the reader post card when blogName is empty

planarvoid committed:
Merge branch 'develop' into feature/12641-add-pexel-source-to-media-picker

planarvoid committed:
Update fluxc

mzorz committed:
Merge pull request #13006 from wordpress-mobile/feature/stories-analytics-interface

WP Stories: Handle analytics events from library

ashiagr committed:
Fix occasional table not found exception on refresh

Delete records instead of tbl drop + create

ashiagr committed:
Fix occasional table not found exception on refresh

Delete records instead of tbl drop + create

loremattei committed:
Update release notes for 15.9

loremattei committed:
Update metadata strings

malinajirka committed:
Merge pull request #13073 from wordpress-mobile/issue/13016-fix-no-such-table-tbl-discover-cards

Fix SQLiteException: no such table: tbl_discover_cards

planarvoid committed:
Merge pull request #13009 from wordpress-mobile/feature/12641-add-pexel-source-to-media-picker

Add stock media source to media picker

planarvoid reviewed (approved):
Thanks for all the changes! It's looking really great, good job 👍 

It's looking good and works really well! I have a few comments for the code.


> Keep in mind that the functionality for marking a child as spam still exists, I just deactivated through the removal of onMenuAction. It will be easier to change the behavior if we want to, but I am happy to completely rollback these changes if you think it's better that way.

This is more than enough 👍 

> I was wrong in thinking that the behavior was caused by some groups having an URL and others don't. Indeed we have groups with and without URLs, but it has nothing to do but it had no correlation with the behavior. I noticed empirically that all groups that can be marked as spam can be resolved to a valid URL, so I added a validation using UrlUtils.isValidUrlAndHostNotNull("https://${group.name}") and a small modification in FluxC to replicate this behavior. It's a hack, but it's working exactly like in wordpress.com. Let me know what you think about it.

Good catch 👍 , it's working well.

marecar3 committed:
Set initial gutenberg props builder as latest

marecar3 committed:
Rename mLatestGutenbergPropsBuilder to mCurrentGutenbergPropsBuilder for better naming convention

marecar3 committed:
Merge pull request #13076 from wordpress-mobile/fix/2692_Unsupported_Block_Editor_cannot_save_changes

khaykov committed:
Merge pull request #13065 from wordpress-mobile/issue/13064-jetpack-site-users

Login: Update behavior when logging in with a Jetpack site so it defaults to using site credentials instead of WP.com credentials

loremattei committed:
Merge pull request #13074 from wordpress-mobile/release/15.9

Merge release notes for Release/15.9 into develop

loremattei committed:
Merge pull request #13083 from wordpress-mobile/fix/cherry-pick-no-such-table-sql-exception

Fix occasional table not found exception on refresh

loremattei committed:
Bump version for 15.8.1

loremattei committed:
Fix merge conflicts

loremattei committed:
Merge pull request #13084 from wordpress-mobile/merge-hotfix-15.8.1-into-15.9

Merge hotfix 15.8.1 into 15.9

loremattei committed:
Bump version to 15.9-rc-2/alpha-248

jkmassel committed:
Merge pull request #13088 from wordpress-mobile/release/15.9

Merge Release 15.9-rc-2 into develop

marco-altran committed:
Bumped FluxC with proper tag

marco-altran committed:
Merge pull request #3 from wordpress-mobile/develop

Rebase

marco-altran committed:
Merge branch 'develop' into feature/mark-referrers-as-spam

# Conflicts:
#	build.gradle

planarvoid reviewed (approved):
It looks good and works well! Thanks for the updates 👍
